### PR TITLE
Win build to create cp2cp during CI

### PR DESCRIPTION
**Pull request by fvacek**

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -1,39 +1,48 @@
...
+      uses: actions-rust-lang/setup-rust-toolchain@v1
+      with:
+        target: ${{ matrix.target }}
+        toolchain: ${{ matrix.toolchain }}
+        override: true
+
+    - name: Build and Install Binary
+      run: |
+        mkdir -p install_dir
+        cargo install --path . --root install_dir --target ${{ matrix.target }}
```
Comment (syyyr) [inline .github/workflows/rust.yml:40]: If you mean for the path to be the git root dir, use:
```suggestion
        cargo install --path  {{github.workspace}} --root install_dir --target ${{ matrix.target }}
```
Also, ideally don't use any relative paths whatsoever (so, --root should have an absolute path and mdir also, you can base all of them off f {{github.workspace}}

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -1,39 +1,48 @@
...
+      - win-build
   pull_request:
-    branches: [ "master" ]
-
-env:
-  CARGO_TERM_COLOR: always
-  RUSTFLAGS: "-Dwarnings"
+    branches:
+      - main
+      - win-build
```
Comment (syyyr) [inline .github/workflows/rust.yml:11]: Are the win-build branches needed?

File: .github/workflows/rust.yml [hunk 3]
```diff
@@ -1,39 +1,48 @@
...
+      with:
+        target: ${{ matrix.target }}
+        toolchain: ${{ matrix.toolchain }}
+        override: true
+
+    - name: Build and Install Binary
+      run: |
+        mkdir -p install_dir
+        cargo install --path . --root install_dir --target ${{ matrix.target }}
+        ls -l install_dir/bin
```
Comment (syyyr) [inline .github/workflows/rust.yml:41]: Is this a remnant from debugging?

File: .github/workflows/rust.yml [hunk 4]
```diff
@@ -1,39 +1,48 @@
...
+      shell: bash
+
+    - name: Upload Artifact
+      uses: actions/upload-artifact@v4
+      with:
+        name: ${{ matrix.os }}-${{ matrix.toolchain }}-binary
+        path: install_dir/bin
```
Comment (syyyr) [inline .github/workflows/rust.yml:48]: Use an absolute path here too please

File: .github/workflows/rust.yml [hunk 5]
```diff
@@ -1,39 +1,48 @@
-name: Rust
+name: Rust Build and Artifact Creation
```
Comment (syyyr) [inline .github/workflows/rust.yml:1]: nit: I'm not sure if "Build and Artifact Creation" really gives any relevant info... I'd just say "Build". Rust is obvious, this is a Rust project and artifacts are created more or less as a side effect.

**Review by syyyr — CHANGES_REQUESTED** [review]
I'd also suggest uploading the artifacts as a nightly release as we do in shvspy [here](https://github.com/silicon-heaven/shvspy/blob/master/.github/workflows/build.yml#L185-L212)

Otherwise, I like it, very nice.

File: .github/workflows/rust.yml (hunk 5 above)
Comment (fvacek) [reply .github/workflows/rust.yml:1]: This is what AI though up :)

File: .github/workflows/rust.yml (hunk 2 above)
Comment (fvacek) [reply .github/workflows/rust.yml:11]: We need `cp2cp.exe`

File: .github/workflows/rust.yml (hunk 3 above)
Comment (fvacek) [reply .github/workflows/rust.yml:41]: Remnant from AI :)

File: .github/workflows/rust.yml (hunk 5 above)
Comment (fvacek) [reply .github/workflows/rust.yml:1]: Anyway, it is building and creating artifacts indeed, I suggest to remove `Rust`.

File: .github/workflows/rust.yml (hunk 3 above)
Comment (fvacek) [reply .github/workflows/rust.yml:41]: Why the shell should not be specified?

Comment (syyyr) [reply .github/workflows/rust.yml:41]: I was referring to the `ls -l install_dir/bin`, not the shell:bash line. Github screwed up. Anyway, there's no need to do `ls`

File: .github/workflows/rust.yml (hunk 2 above)
Comment (syyyr) [inline .github/workflows/rust.yml:11]:
```suggestion
```

File: .github/workflows/rust.yml [hunk 6]
```diff
@@ -1,39 +1,48 @@
...
 
 on:
   push:
-    branches: [ "master" ]
+    branches:
+      - main
+      - win-build
```
Comment (syyyr) [inline .github/workflows/rust.yml:7]:
```suggestion
```

File: .github/workflows/rust.yml (hunk 5 above)
Comment (syyyr) [reply .github/workflows/rust.yml:1]: I think artifacts are a side effect of a build. Every build will have artifacts (whether you upload them or not). I don't have a strong opinion though, so change it as you want

**Comment by syyyr** [comment]
fyi: the build fails because you have to (single) quote the github.workspace thingy

File: .github/workflows/rust.yml (hunk 5 above)
Comment (fvacek) [reply .github/workflows/rust.yml:1]: So what about `Build and Artifact Upload`?

Comment (syyyr) [reply .github/workflows/rust.yml:1]: It's not better for me, as I don't think it's necessary to include "artifacts" in the name, but I don't really care :D name it whatever you want. It was just nitpick anyway

**Comment by fvacek** [comment]
> I'd also suggest uploading the artifacts as a nightly release as we do in shvspy [here](https://github.com/silicon-heaven/shvspy/blob/master/.github/workflows/build.yml#L185-L212)
> 
> Otherwise, I like it, very nice.

Yes, to make release would be nice. I would better prefer to do it in separate PR.

File: .github/workflows/rust.yml (hunk 3 above)
Comment (fvacek) [reply .github/workflows/rust.yml:41]: Sure, AI remnant.

File: .github/workflows/rust.yml (hunk 2 above)
Comment (fvacek) [reply .github/workflows/rust.yml:11]: How can I run test of this PR without `win_build`?

Comment (syyyr) [reply .github/workflows/rust.yml:11]: Is it even necessary? I think the pr: main section should do the job... Anyway, I guess you can also remove it after merging the pr

**Review by syyyr — APPROVED** [review]

**Pull request merged**